config/connections.go

```diff
@@ -123,7 +123,10 @@ func CreateHubConnection(
 		return nil, nil, err
 	}
 
-	honoConfig.CleanSession = true
+	if len(settings.ClientID) == 0 {
```

Comment:
I am not sure that this change will fix something. According to [the Hono specification](https://www.eclipse.org/hono/docs/user-guide/mqtt-adapter/), session state is not maintained.

Reply:
Session state is ignored by hono but not by paho client. As a result all unsent messages will be dropped.